### PR TITLE
ci: Upload stacker binary

### PR DESCRIPTION
**Pull request by smoser**
Uploading a c-i built binary can make it easier for users to test.

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by smoser** [comment]
You can see the 'binary' and 'binary-cov' artifacts https://github.com/project-stacker/stacker/actions/runs/6932027516?pr=556 (scroll down, 'Artifacts')

**Comment by hallyn** [comment]
Stupid question: how long will github keep these artifacts?  I assume this is a handled problem, and it has reasonable retention times and auto-purge ....

**Review by hallyn — APPROVED** [review]

File: .github/workflows/coverage.yaml [hunk 1]
```diff
@@ -91,6 +91,14 @@ jobs:
           REGISTRY_URL: localhost:5000
           ZOT_HOST: localhost
           ZOT_PORT: 8080
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        if: ${{ matrix.privilege-level == 'priv' }}
+        with:
+          # if there is more than 1 go-version, we would need to account for that here.
+          name: binary-cov
```
Comment (rchincha) [inline .github/workflows/coverage.yaml:99]: same here?

**Comment by smoser** [comment]
> Stupid question: how long will github keep these artifacts? I assume this is a handled problem, and it has reasonable retention times and auto-purge ...

90 days by default [[1](https://docs.github.com/en/organizations/managing-organization-settings/configuring-the-retention-period-for-github-actions-artifacts-and-logs-in-your-organization)]:

> By default, the artifacts and log files generated by workflows are retained for 90 days before they are automatically deleted. You can adjust the retention period, depending on the type of repository:
>
> * For public repositories: you can change this retention period to anywhere between 1 day or 90 days.
> * For private repositories: you can change this retention period to anywhere between 1 day or 400 days.
> 
> When you customize the retention period, it only applies to new artifacts and log files, and does not retroactively apply to existing objects. For managed repositories and organizations, the maximum retention period cannot exceed the limit set by the managing organization or enterprise.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -84,6 +84,14 @@ jobs:
           REGISTRY_URL: localhost:5000
           ZOT_HOST: localhost
           ZOT_PORT: 8080
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        if: ${{ matrix.privilege-level == 'priv' }}
+        with:
+          # if there is more than 1 go-version, we would need to account for that here.
+          name: binary
```
[the remark this answers is not in the document]
Comment (smoser) [reply .github/workflows/build.yaml:92]: If you want we can call it stacker. but what that creates is a zip file name 'stacker.zip' which contains stacker.  I'd prefer 'binaries' or 'binary' as its an archive (potentially multiple things). 

So its not like it is a single thing that that you can download it and use it, in which case 'stacker' and 'stacker-cov' would be the right name.

I'll rename it if you want though.

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**